presto-jdbc/src/main/java/io/prestosql/jdbc/AbstractPrestoResultSet.java

```diff
@@ -649,7 +649,10 @@ private static Object convertFromClientRepresentation(ClientTypeSignature column
             }
         }
 
-        // TODO (https://github.com/prestosql/presto/issues/6048) add conversions for decimal, date, time, timestamp, interval
+        Class<?> defaultRepresentation = DEFAULT_OBJECT_REPRESENTATION.get(columnType.getRawType());
+        if (defaultRepresentation != null) {
+            return TYPE_CONVERSIONS.convert(columnType, value, defaultRepresentation);
```

Comment:
is the "else" code path needed only for "plugin types" ?

Reply:
No, i did not cover all the builtin types